### PR TITLE
[libc] Implement process_mrelease

### PR DESCRIPTION
**Pull request by moar55**
This PR implements process_mrelease.
A previous PR was merged #117503, but failed on merge due to an issue in the tests. Namely the failing tests were comparing against return type as opposed to errno. This is fixed in this PR.

### CODE REVIEW

**Comment by moar55** [comment]
@SchrodingerZhu hey can you re-review this :)

**Comment by SchrodingerZhu** [comment]
@moar55 it should be the case that the header is too long. You can shrink the license header by removing some `-`

**Comment by moar55** [comment]
@SchrodingerZhu hey it passes now, can you double-check all is good :)

File: libc/test/src/sys/mman/linux/CMakeLists.txt [hunk 1]
```diff
@@ -181,3 +181,23 @@ add_libc_unittest(
...
+    libc.src.errno.errno
+    libc.src.sys.mman.process_mrelease
+    libc.src.unistd.close
+    libc.src.signal.kill
+    libc.include.signal
+    libc.src.stdlib.exit
+    libc.src.__support.OSUtil.osutil
+    libc.src.__support.threads.sleep
+    libc.test.UnitTest.ErrnoSetterMatcher
+    )
```
Comment (SchrodingerZhu) [inline libc/test/src/sys/mman/linux/CMakeLists.txt:203]: can u remove the leading space on line 203

**Review by SchrodingerZhu — APPROVED** [review]
LGTM with a small nit.

File: libc/test/src/sys/mman/linux/process_mrelease_test.cpp [hunk 2]
```diff
@@ -0,0 +1,74 @@
...
+    // Send SIGKILL to child process
+    LIBC_NAMESPACE::kill(child_pid, SIGKILL);
+
+    EXPECT_THAT(LIBC_NAMESPACE::process_mrelease(pidfd, 0), Succeeds());
+
+    LIBC_NAMESPACE::close(pidfd);
+  }
+}
+
+TEST(LlvmLibcMProcessMReleaseTest, ErrorNotKilled) {
```
Comment (SchrodingerZhu) [inline libc/test/src/sys/mman/linux/process_mrelease_test.cpp:52]: There seems to an extra `M` in name.

**Comment by moar55** [comment]
@SchrodingerZhu hey can you merge this,?
I have addressed the comments.

**Review by SchrodingerZhu — APPROVED** [review]

**Pull request merged**